### DISCUSSION TITLE
v3.0: PostCSS plugin tailwindcss requires PostCSS 8

### DISCUSSION THREAD

**Discussion by sifferhans** [tailwindlabs/tailwindcss#6337] [Help] [answered]
I wanted to migrate my project to **Tailwind 3.0**, and I installed PostCSS 8. I still get this error.
Any ideas what might be the problem? 

Here are all dependencies I use

```
"dependencies": {
    "@headlessui/vue": "^1.4.1",
    "@heroicons/vue": "^1.0.5",
    "@stripe/stripe-js": "^1.21.1",
    "apexcharts": "^3.31.0",
    "firebase": "^9.5.0",
    "i18n-iso-countries": "^6.8.0",
    "idb": "^6.1.5",
    "isomorphic-fetch": "^3.0.0",
    "opensheetmusicdisplay": "^1.3.1",
    "plyr": "^3.6.9",
    "regenerator-runtime": "^0.13.9",
    "showdown": "^1.9.1",
    "unzipper": "^0.10.11",
    "vue": "^3.2.23",
    "vue-class-component": "^8.0.0-0",
    "vue-i18n": "^9.1.9",
    "vue-router": "^4.0.12",
    "vue3-apexcharts": "^1.4.1",
    "vuedraggable": "^4.1.0",
    "vuex": "^4.0.2"
  },
  "devDependencies": {
    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
    "@crowdin/crowdin-api-client": "^1.12.3",
    "@tailwindcss/forms": "^0.4.0",
    "@tsconfig/node14": "^1.0.1",
    "@types/chai": "^4.2.22",
    "@types/isomorphic-fetch": "0.0.35",
    "@types/mocha": "^5.2.4",
    "@types/node": "^14.17.34",
    "@types/showdown": "^1.9.4",
    "@types/unzipper": "^0.10.4",
    "@typescript-eslint/eslint-plugin": "^2.33.0",
    "@typescript-eslint/parser": "^2.33.0",
    "@vue/cli-plugin-babel": "^4.5.15",
    "@vue/cli-plugin-eslint": "^4.5.15",
    "@vue/cli-plugin-router": "^4.5.15",
    "@vue/cli-plugin-typescript": "^4.5.15",
    "@vue/cli-plugin-unit-mocha": "^4.5.15",
    "@vue/cli-plugin-vuex": "^4.5.15",
    "@vue/cli-service": "^4.5.15",
    "@vue/compiler-sfc": "^3.2.23",
    "@vue/eslint-config-typescript": "^5.1.0",
    "@vue/test-utils": "^2.0.0-rc.17",
    "autoprefixer": "^10.4.0",
    "chai": "^4.3.4",
    "eslint": "^6.7.2",
    "eslint-plugin-vue": "^7.20.0",
    "firebase-tools": "^9.23.0",
    "node-sass": "^4.12.0",
    "postcss": "^8.4.4",
    "sass-loader": "^8.0.2",
    "tailwindcss": "^3.0.0",
    "tslib": "^2.3.1",
    "typescript": "^4.5.2",
    "vue-cli-plugin-tailwind": "~2.0.6"
  }
```

**Comment by magick2** [comment]
Same problem. Could you solve it?

**Comment by lukewarlow** [comment]
You need to update to Vue CLI 5 (currently in RC) to use postcss 8 and in turn Tailwind v3.

**Reply by sifferhans** [reply]
I already have. it doesn't make a difference unfortunately

**Reply by lukewarlow** [reply]
Check your package-lock and see which package is relying on an old version of postcss. Somewhere your project is using postcss 7 not 8.

**Reply by wutzi15** [reply]
I have the same problem. I can confirm that `yarn global add @vue/cli@next` fixes the problem.

**Reply by bencemarki** [reply]
I had the same problem but after installing cli 5.0.0-rc.1 i got the following error when I try to start my dev server with npm run serve and I couldnt find a solution for it yet:

 INFO  Starting development server...
 ERROR  TypeError: webpack.EntryPlugin is not a constructor
TypeError: webpack.EntryPlugin is not a constructor
    at serve (C:\Munka\noiegeszseg\noiegeszseg-frontend\node_modules\@vue\cli-service\lib\commands\serve.js:164:9)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Process finished with exit code 1

Any idea what could go wrong?

**Reply by lukewarlow** [reply]
I suggest checking out the Vue CLI upgrade guide or taking any queries to their discussions or equivalent place to get the best help

**Reply by sifferhans** [reply]
I got this working by also updating all @vue/cli plugins to the newest version

**Comment by BkunS** [comment]
Same problem after going through setup of tailwind css 3.0 with create-react-app. Shows the error: `Error: PostCSS plugin tailwindcss requires PostCSS 8.`

Node: v14.17.0
```
"devDependencies": {
    "autoprefixer": "^10.4.0",
    "postcss": "^8.4.5",
    "tailwindcss": "^3.0.5"
}
```

**Reply by lukewarlow** [reply]
Which version of create react app are you using? As far as I know CRA still uses postcss 7 under the hood. Unless you're using the alpha version.

**Reply by BkunS** [reply]
I did solve the issue by getting help from [here](https://github.com/tailwindlabs/tailwindcss/discussions/6472). Latest CRA (v5.0) works, and try NOT to use any templates.

**Reply by lukewarlow** [reply]
Alternatively for any other React users Adam has put together a [demo repo](https://github.com/adamwathan/cra-tailwind-3) where he uses the CLI.

**Comment by sifferhans** [comment, marked as answer]
I got it working!
I had to update the @vue/cli and all its plugins to the newest version (5.0.0-rc.1), as the versions I had installed used PostCSS 7.

**Reply by matthew-hayden** [reply]
Thank you for this! Worked for me as well.
